src/common/types/scalars/IP.js

```
@@ -0,0 +1,20 @@
+import { GraphQLScalarType, Kind } from 'graphql';
+import isIP from 'validator/lib/isIP';
+
+// Exported for testing
+export const maybeIP4 = val => (isIP(val, 4) ? val : null);
```

The only reason why I wanted custom scalar type is the fact that you can do quick validations and improve DX. Therefore you should throw an exception here. Now it's still possible to call `geoIP` query with whatever you want and the result is this call: `https://geoip-api-prod.skypicker.com/geoip-api?ip=null` (GeoIP server cannot handle it).

This is what you'll get with `Error` instead of `null`:

```json
{
  "errors": [
    {
      "message": "Expected type IP!, found \"x\"; Value is not a valid IP address",
      "locations": [
        {
          "line": 2,
          "column": 13
        }
      ]
    }
  ],
}
```

And this is what you'll get with current code:

```json
{
  "errors": [
    {
      "message": "Bad Gateway",
      "locations": [
        {
          "line": 2,
          "column": 3
        }
      ],
      "path": [
        "geoIP"
      ],
      "_proxy": {
        "statusCode": 502,
        "url": "https://geoip-api-prod.skypicker.com/geoip-api?ip=null"
      }
    }
  ],
  "data": {
    "geoIP": null
  },
}
```